include/tsl/ordered_hash.h

```diff
@@ -1098,6 +1098,44 @@ class ordered_hash : private Hash, private KeyEqual {
     return 1;
   }
 
+  /**
+   * Clear a bucket without touching the container holding the values.
+   */
+  void clear_bucket(typename buckets_container_type::iterator it) {
+    tsl_oh_assert(it != m_buckets_data.end());
+    it->clear();
+    backward_shift(std::size_t(std::distance(m_buckets_data.begin(), it)));
+  }
+
+  /**
+   * Remove all entries for which the given predicate matches.
+   */
+  template <class Predicate>
+  size_type erase_if(Predicate &pred) {
+    using const_ref = typename values_container_type::const_reference;
+    auto last = m_values.end();
+    auto first = m_values.begin();
+    for (; first != last; ++first) {
+      if (pred(static_cast<const_ref>(*first))) {
+        clear_bucket(find_key(KeySelect()(*first), hash_key(KeySelect()(*first))));
+        for (auto it = first; ++it != last; ) {
+          auto it_bucket = find_key(KeySelect()(*it), hash_key(KeySelect()(*it)));
+          if (pred(static_cast<const_ref>(*it))) {
+            clear_bucket(it_bucket);
+          }
+          else {
+            it_bucket->set_index(static_cast<index_type>(std::distance(m_values.begin(), first)));
```

Comment:
Don't forget to format with `clang-format --style=Google` (I need to add a .clang-format to the repo).

Reply:
Done.